### PR TITLE
feat: add api version validation

### PR DESCRIPTION
**Pull request by maliroteh-sf**
### What does this PR do?
Ensure that the org API version is supported by the local dev server.

### What issues does this PR fix or reference?
@W-16974125@

### CODE REVIEW

File: messages/shared.utils.md [hunk 1]
```diff
@@ -29,3 +29,11 @@ You must provide valid SSL certificate data
...
 
 Local Dev is not enabled for your org. See https://developer.salesforce.com/docs/platform/lwc/guide/get-started-test-components.html for more information on enabling and using Local Dev.
+
+# error.org.api-mismatch.message
+
+Your org is on API version %s but this CLI plugin supports API version %s.
+
+# error.org.api-mismatch.remediation
+
+Please reinstall or update the plugin using %s tag.
```
Comment (maliroteh-sf) [inline messages/shared.utils.md:39]: I'm open to rewording these if you have any recommendations.

File: package.json [hunk 2]
```diff
@@ -215,6 +215,23 @@
       "output": []
     }
   },
+  "apiVersionMetadata": {
```
Comment (maliroteh-sf) [inline package.json:218]: In orgUtils.ts I have provided details on what this section is and how it is used

File: .husky/check-versions.cjs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const fs = require('fs');
+
+// Read package.json
+const packageJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+
+// Extract versions
```
[the remark this answers is not in the document]
Comment (khawkins) [reply .husky/check-versions.cjs:6]: Nit: does it have to be CommonJS? All of our supported versions of node support EcmaScript modules.

File: src/shared/orgUtils.ts [hunk 4]
```diff
@@ -18,6 +24,39 @@ export type AppDefinition = {
...
+ * to the same version otherwise the pre-commit will fail. This means that, as the PR owner deliberately
+ * updates "target.matchingDevServerVersion", they are responsible to ensuring that the rest of the data under
+ * "apiVersionMetadata" is accurate.
+ *
+ * The "versionToTagMappings" section will provide a mapping between supported API version by the dev server
+ * and the tagged version of our plugin. We use "versionToTagMappings" to convey to the user which version of
+ * our plugin should they be using to match with the API version of their org (i.e which version of our plugin
+ * contains the lwc-dev-server dependency that can support the API version of their org).
+ */
+type apiVersionMetadata = {
```
Comment (khawkins) [inline src/shared/orgUtils.ts:47]: We should case this type properly, as `ApiVersionMetadata`.

File: .husky/check-versions.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  console.error('Error: missing @lwc/lwc-dev-server or matchingDevServerVersion');
+  process.exit(1); // Fail the check
+}
+
+// Compare versions
+if (devServerDependencyVersion === devServerTargetVersion) {
+  process.exit(0); // Pass the check
+} else {
+  console.error(
+    `Error: Versions do not match. @lwc/lwc-dev-server: ${devServerDependencyVersion}, matchingDevServerVersion: ${devServerTargetVersion}`
```
Comment (khawkins) [inline .husky/check-versions.js:20]: I would make this error message more reflective of the process request being conveyed here:
```suggestion
    `Error: @lwc/lwc-dev-server versions do not match between 'dependencies' and 'apiVersionMetadata'. Expected ${devServerDependencyVersion} in apiVersionMetadata > target > matchingDevServerVersion. Got ${devServerTargetVersion} instead. When updating the @lwc/lwc-dev-server dependency, you must ensure that it's compatible with the supported API version in this branch, then update apiVersionMetadata > target > matchingDevServerVersion to match, in order to "sign off" on this dependency change.`
```

File: messages/shared.utils.md [hunk 6]
```diff
@@ -29,3 +29,11 @@ You must provide valid SSL certificate data
 # error.localdev.not.enabled
 
 Local Dev is not enabled for your org. See https://developer.salesforce.com/docs/platform/lwc/guide/get-started-test-components.html for more information on enabling and using Local Dev.
+
+# error.org.api-mismatch.message
+
+Your org is on API version %s but this CLI plugin supports API version %s.
```
Comment (khawkins) [inline messages/shared.utils.md:35]:
```suggestion
Your org is on API version %s, but this version of the CLI plugin supports API version %s.
```

File: messages/shared.utils.md (hunk 1 above)
Comment (khawkins) [inline messages/shared.utils.md:39]:
```suggestion
To use the plugin with this org, you can reinstall or update the plugin using the "%s" tag. For example: "sf plugins install %s".
```

Where the first `%s` would just be the tag name and the second would be the expansion that you currently have in the code, for this text update.

**Review by khawkins — APPROVED** [review]
Looks fine over all. 👍

File: .husky/check-versions.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import fs from 'node:fs';
```
Comment (ravijayaramappa) [inline .husky/check-versions.js:1]: Nice! Thanks.

**Pull request merged**